desc/objectives/_geometry.py

```diff
@@ -750,24 +770,115 @@ def compute(self, equil_params, surface_params=None, constants=None):
             transforms=constants["equil_transforms"],
             profiles=constants["equil_profiles"],
         )
-        plasma_coords = rpz2xyz(jnp.array([data["R"], data["phi"], data["Z"]]).T)
+        plasma_coords_rpz = jnp.array([data["R"], data["phi"], data["Z"]]).T
+        plasma_coords = rpz2xyz(plasma_coords_rpz)
         if self._surface_fixed:
-            surface_coords = constants["surface_coords"]
+            data_surf = constants["data_surf"]
         else:
-            surface_coords = compute_fun(
+            data_surf = compute_fun(
                 self._surface,
                 self._surface_data_keys,
                 params=surface_params,
                 transforms=constants["surface_transforms"],
                 profiles={},
                 basis="xyz",
-            )["x"]
-        d = safenorm(plasma_coords[:, None, :] - surface_coords[None, :, :], axis=-1)
+            )
+
+        surface_coords = data_surf["x"]
+        diff_vec = plasma_coords[:, None, :] - surface_coords[None, :, :]
+        d = safenorm(diff_vec, axis=-1)
+
+        point_signs = jnp.ones(surface_coords.shape[0])
+        if self._use_signed_distance:
+            surface_coords_rpz = xyz2rpz(surface_coords)
+
+            def _find_angle_vec(R, Z, Rsurf, Zsurf):
+                # R Z are plasma points,
+                # Rsurf Zsurf are the surface points being checked for whether
+                # or not they are inside the plasma
+
+                # algorithm based off of "An Incremental Angle Point in Polygon Test",
+                # K. Weiler, https://doi.org/10.1016/B978-0-12-336156-1.50012-4
+
+                Rbool = R[:, None] > Rsurf
+                Zbool = Z[:, None] > Zsurf
+                # these are now size (Nplasma, Nsurf)
+                quads = jnp.zeros_like(Rbool)
+                quads = jnp.where(jnp.logical_and(Rbool, Zbool), 0, quads)
```

Comment:
what is `quads`? 

Also it seems like this is checking where `Rbool` and `Zbool` are both true and making those cases 0, otherwise `quads` (which is already all zero) so doesn't that mean everything stays 0?

Reply:
it is checking in which "quadrant" each point on the plasma is in relation to each point on the surface. The algorithm works by counting the number of points in each quadrant and assigning a value based on which quadrant, then adding them up, and that determines if the point is inside of outside the given perimeter.

ah true, we can skip this check, good catch